docs/concepts/projects/sync.md

```diff
@@ -1,44 +1,154 @@
+and synced before invoking the requested command. This ensures the project environment is always
+up-to-date. Similarly, commands which read the lockfile, such as `uv tree`, will automatically
+update it before running.
+
+To disable automatic locking, use the `--locked` option:
 
 ```console
-$ uv lock
+$ uv run --locked ...
 ```
 
-### Exporting the lockfile
+If the lockfile is not up-to-date, an error will be raised instead of updating the lockfile.
 
-If you need to integrate uv with other tools or workflows, you can export `uv.lock` to
-`requirements.txt` format with `uv export --format requirements-txt`. The generated
-`requirements.txt` file can then be installed via `uv pip install`, or with other tools like `pip`.
+To use the lockfile without checking if it is up-to-date, use the `--frozen` option:
 
-In general, we recommend against using both a `uv.lock` and a `requirements.txt` file. If you find
-yourself exporting a `uv.lock` file, consider opening an issue to discuss your use case.
+```console
+$ uv run --frozen ...
+```
 
-### Checking if the lockfile is up-to-date
+Similarly, to run a command without checking if the environment is up-to-date, use the `--no-sync`
+option:
 
-To avoid updating the lockfile during `uv sync` and `uv run` invocations, use the `--frozen` flag.
+```console
+$ uv run --no-sync ...
+```
 
-To avoid updating the environment during `uv run` invocations, use the `--no-sync` flag.
+## Checking if the lockfile is up-to-date
 
-To assert the lockfile matches the project metadata, use the `--locked` flag. If the lockfile is not
-up-to-date, an error will be raised instead of updating the lockfile.
+When considering if the lockfile is up-to-date, uv will check if it matches the project metadata.
+For example, if you add a dependency to your `pyproject.toml`, the lockfile will be outdated.
+Similarly, if you change the version constraints for a dependency such that the locked version is
+excluded, the lockfile will be outdated. However, if you change the version constraints such that
+the existing locked version is still included, the lockfile will still be considered up-to-date.
 
-You can also check if the lockfile is up-to-date by passing the `--check` flag to `uv lock`:
+You can check if the lockfile is up-to-date by passing the `--check` flag to `uv lock`:
 
 ```console
 $ uv lock --check
 ```
 
 This is equivalent to the `--locked` flag for other commands.
 
-### Upgrading locked package versions
+!!! important
+
+    uv will not consider lockfiles outdated when new versions of packages are released — the lockfile
+    needs to be explicitly updated if you want to upgrade dependencies. See the documentation on
+    [upgrading locked package versions](#upgrading-locked-package-versions) for details.
+
+## Creating the lockfile
+
+While the lockfile is created [automatically](#automatic-lock-and-sync), the lockfile may also be
```

Comment:
```suggestion
While the lockfile is created [automatically](#automatic-lock-and-sync) via `uv run`, the lockfile may also be
```

Reply:
And `uv sync` and `uv tree` etc.? I don't think I want to specify that here